File: src/main/java/com/dambra/paul/stringcalculator/calculationParts/CalculationPart.java
package com.dambra.paul.stringcalculator.calculationParts;

public abstract class CalculationPart {
    protected CalculationPart next;

    public abstract double Calculate(double total);

    public CalculationPart addNext(CalculationPart next) {
        this.next = next;
        return this.next;
    }
}

File: src/main/java/com/dambra/paul/stringcalculator/calculationParts/CalculationPart.kt
package com.dambra.paul.stringcalculator.calculationParts

abstract class CalculationPart {
    protected lateinit var next: CalculationPart

    abstract fun calculate(total: Double): Double

    fun addNext(next: CalculationPart): CalculationPart {
        this.next = next
        return this.next
    }
}

